### PR TITLE
[MRG] avoid inplace changes when passing events

### LINKED ISSUE #401: Write -> Read -> Write -> ... cycles lead do decimation of events
#### Describe the bug
Repeatedly writing and reading the same dataset decimates the events present in the data.

#### Steps to reproduce
```python
import pathlib
from tempfile import TemporaryDirectory
import mne
from mne_bids import make_bids_basename, read_raw_bids, write_raw_bids


mne.set_log_level(verbose=False)


def write_and_read_bids(raw, events, subjcect, bids_root):
    bids_basename = make_bids_basename(subject=subject)
    bids_fname = bids_basename + '_meg.fif'

    print(f'Number of events before writing: {events.shape[0]}')
    write_raw_bids(raw=raw,
                   bids_basename=bids_basename,
                   bids_root=bids_root,
                   events_data=events,
                   overwrite=True,
                   verbose=False)

    raw_read = read_raw_bids(bids_fname=bids_fname,
                             bids_root=bids_root,
                             verbose=False)
    print(f'Number of Annotations after reading: {len(raw_read.annotations)}')



subject = 'sample'
data_dir = pathlib.Path(mne.datasets.sample.data_path())
raw_fname = data_dir / 'MEG' / subject / 'sample_audvis_raw.fif'
raw = mne.io.read_raw_fif(raw_fname)
raw_copy = raw.copy()  # Gonna use this for a small test later.
events = mne.find_events(raw)


with TemporaryDirectory() as bids_root:
    for pass_count in range(1, 5+1):
        print(f'\n==> Pass number {pass_count} …')
        write_and_read_bids(raw=raw, events=events, subjcect=subject,
                            bids_root=bids_root)


# Check that raw hasn't been modified in-place
raw.load_data()
raw_copy.load_data()
print(f'\n==> raw == raw_copy: {raw == raw_copy}')
```

#### Expected results
```
==> Pass number 1 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> Pass number 2 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> Pass number 3 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> Pass number 4 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> Pass number 5 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> raw == raw_copy: True
```

#### Actual results
```
==> Pass number 1 …
Number of events before writing: 320
Number of Annotations after reading: 320

==> Pass number 2 …
Number of events before writing: 320
Number of Annotations after reading: 260

==> Pass number 3 …
Number of events before writing: 320
Number of Annotations after reading: 194

==> Pass number 4 …
Number of events before writing: 320
Number of Annotations after reading: 130

==> Pass number 5 …
Number of events before writing: 320
Number of Annotations after reading: 65

==> raw == raw_copy: True
```

#### Additional information
`mne-bids:master`
`mne-python` 0.20.3

### PR DESCRIPTION
**Pull request by agramfort**
closes #401

@hoechenberger ok for you?

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by jasmainak** [comment]
LGTM, do we need a test?

**Comment by hoechenberger** [comment]
@jasmainak I've added a test

File: mne_bids/tests/test_write.py [hunk 1]
```diff
@@ -1159,3 +1159,22 @@ def test_write_anat_pathlike():
...
     assert isinstance(anat_dir, str)
+
+
+def test_write_does_not_alter_events_inplace():
+    """Test that writing does not modify the passed events array."""
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    events_fname = (Path(data_path) / 'MEG' / 'sample' /
+                    'sample_audvis_trunc_raw-eve.fif')
```
Comment (jasmainak) [inline mne_bids/tests/test_write.py:1170]: why not `op.join` ?

**Comment by jasmainak** [comment]
Other than minor nitpick, looks good to me. Feel free to merge once CIs are green.

File: mne_bids/tests/test_write.py (hunk 1 above)
Comment (hoechenberger) [reply mne_bids/tests/test_write.py]: Whoops, leftover from copying over from another test. changed that.

**Comment by agramfort** [comment]
@hoechenberger any chance to rather update an existing test to limit testing time ?

**Comment by hoechenberger** [comment]
> @hoechenberger any chance to rather update an existing test to limit testing time ?

I could but is it really worth it? Tradeoff is always that the tests get quite convoluted…

```
0.69s call     mne_bids/tests/test_write.py::test_write_does_not_alter_events_inplace
```
on `drago`'s NAS

**Comment by agramfort** [comment]
Ok then

**Pull request merged**